If publishing one package fails the whole process crashes

```
Created the release: https://github.com/ckeditor/ckeditor5-build-decoupled-document/releases/tag/v1.0.0-beta.2
Creating a GitHub release for "@ckeditor/ckeditor5-build-inline"...
Created the release: https://github.com/ckeditor/ckeditor5-build-inline/releases/tag/v1.0.0-beta.2
Creating a GitHub release for "@ckeditor/ckeditor5-clipboard"...
Created the release: https://github.com/ckeditor/ckeditor5-clipboard/releases/tag/v1.0.0-beta.2
Creating a GitHub release for "@ckeditor/ckeditor5-cloud-services"...
<!DOCTYPE html>
<!--

Hello future GitHubber! I bet you're here to remove those nasty inline styles,
DRY up these templates and make 'em nice and re-usable, right?

Please, don't. https://github.com/styleguide/templates/2.0

-->
<html>
  <head>
    <meta http-equiv="Content-type" content="text/html; charset=utf-8">
    <title>Unicorn! &middot; GitHub</title>
    <style type="text/css" media="screen">
      body {
        background-color: #f1f1f1;
        margin: 0;
        font-family: "Helvetica Neue", Helvetica, Arial, sans-serif;
      }

      .container { margin: 50px auto 40px auto; width: 600px; text-align: center; }

      a { color: #4183c4; text-decoration: none; }
      a:hover { text-decoration: underline; }

      h1 { letter-spacing: -1px; line-height: 60px; font-size: 60px; font-weight: 100; margin: 0px; text-shadow: 0 1px 0 #fff; }
      p { color: rgba(0, 0, 0, 0.5); margin: 10px 0 10px; font-size: 18px; font-weight: 200; line-height: 1.6em;}

      ul { list-style: none; margin: 25px 0; padding: 0; }
      li { display: table-cell; font-weight: bold; width: 1%; }

      .logo { display: inline-block; margin-top: 35px; }
      .logo-img-2x { display: none; }
      @media
      only screen and (-webkit-min-device-pixel-ratio: 2),
      only screen and (   min--moz-device-pixel-ratio: 2),
      only screen and (     -o-min-device-pixel-ratio: 2/1),
      only screen and (        min-device-pixel-ratio: 2),
      only screen and (                min-resolution: 192dpi),
      only screen and (                min-resolution: 2dppx) {
        .logo-img-1x { display: none; }
        .logo-img-2x { display: inline-block; }
      }

      #suggestions {
        margin-top: 35px;
        color: #ccc;
      }
      #suggestions a {
        color: #666666;
        font-weight: 200;
        font-size: 14px;
        margin: 0 10px;
      }

    </style>
  </head>
  <body>

    <div class="container">
      <p>
        <img width="200" src="...">
      </p>

      <p><strong>We had issues producing the response to your request.</strong></p>
      <p>Sorry about that. Please try refreshing and contact us if the problem persists.</p>
      <div id="suggestions">
        <a href="https://github.com/contact">Contact Support</a> &mdash;
        <a href="https://status.github.com">GitHub Status</a> &mdash;
        <a href="https://twitter.com/githubstatus">@githubstatus</a>
      </div>

      <a href="/" class="logo logo-img-1x">
        <img width="32" height="32" title="" alt="" src="..." />
      </a>

      <a href="/" class="logo logo-img-2x">
        <img width="32" height="32" title="" alt="" src="..." />
      </a>
    </div>
  </body>
</html>
npm ERR! code ELIFECYCLE
npm ERR! errno 255
npm ERR! ckeditor5@1.0.0-beta.1 release:dependencies: `node ./scripts/release/release-dependencies.js`
npm ERR! Exit status 255
npm ERR!
npm ERR! Failed at the ckeditor5@1.0.0-beta.1 release:dependencies script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/p/.npm/_logs/2018-04-10T08_59_12_807Z-debug.log
(master c9097cf) p@m /workspace/ckeditor5>
```

This means that releases for all other packages weren't published to GH. And that kinda doesn't make sense. Now we have to clean this up manually for all remaining packages. It'd be better if only this one was a problem.

--- a/packages/ckeditor5-dev-env/lib/release-tools/tasks/releasesubrepositories.js
+++ b/packages/ckeditor5-dev-env/lib/release-tools/tasks/releasesubrepositories.js
@@ -275,13 +275,21 @@ module.exports = function releaseSubRepositories( options ) {
 			};
 
 			return createGithubRelease( releaseOptions.token, githubReleaseOptions )
-				.then( () => {
-					// eslint-disable-next-line max-len
-					const url = `https://github.com/${ repositoryInfo.owner }/${ repositoryInfo.name }/releases/tag/v${ releaseDetails.version }`;
-					log.info( `Created the release: ${ url }` );
+				.then(
+					() => {
+						// eslint-disable-next-line max-len
+						const url = `https://github.com/${ repositoryInfo.owner }/${ repositoryInfo.name }/releases/tag/v${ releaseDetails.version }`;
+						log.info( `Created the release: ${ url }` );
 
-					return Promise.resolve();
-				} );
+						return Promise.resolve();
+					},
+					err => {
+						log.info( 'Cannot create a release on GitHub. Skipping that package.' );
+						log.error( err );
+
+						return Promise.resolve();
+					}
+				);
 		} );
 	}
 

--- a/packages/ckeditor5-dev-env/tests/release-tools/tasks/releasesubrepositories.js
+++ b/packages/ckeditor5-dev-env/tests/release-tools/tasks/releasesubrepositories.js
@@ -547,6 +547,87 @@ describe( 'dev-env/release-tools/tasks', function() {
 					expect( stubs.logger.error.getCall( 4 ).args[ 0 ] ).to.equal( '* Cannot find changelog entry for version 0.2.1.' );
 				} );
 		} );
+
+		it( 'error during publishing on github does not break the process', () => {
+			const options = {
+				cwd: mainPackagePath,
+				packages: 'packages'
+			};
+
+			makeCommit( 'alpha', 'Feature: This is an initial commit.' );
+			makeChangelog( 'alpha', [
+				'Changelog',
+				'=========',
+				'',
+				'## 0.1.0 (2017-06-08)',
+				'',
+				'### Features',
+				'',
+				'* This is an initial commit.',
+				''
+			].join( '\n' ) );
+			makeCommit( 'alpha', 'Docs: Changelog. [skip ci]', [ 'CHANGELOG.md' ] );
+
+			makeCommit( 'beta', 'Fix: Some fix.' );
+			makeChangelog( 'beta', [
+				'Changelog',
+				'=========',
+				'',
+				'## [0.2.1](https://github.com/ckeditor) (2017-06-08)',
+				'',
+				'### Fix',
+				'',
+				'* Some fix.',
+				''
+			].join( '\n' ) );
+			makeCommit( 'beta', 'Docs: Changelog. [skip ci]', [ 'CHANGELOG.md' ] );
+
+			stubs.cli.confirmRelease.returns( Promise.resolve( true ) );
+
+			const token = 'fooToken';
+			const rejectError = new Error( 'Some error.' );
+
+			stubs.cli.configureReleaseOptions.resolves( {
+				token,
+				skipNpm: true,
+				skipGithub: false
+			} );
+
+			stubs.validatePackageToRelease.returns( [] );
+			stubs.createGithubRelease.onFirstCall().rejects( rejectError );
+			stubs.createGithubRelease.onSecondCall().resolves();
+
+			return releaseSubRepositories( options )
+				.then( () => {
+					expect( executedCommand.length ).to.equal( 12 );
+					expect( stubs.createGithubRelease.callCount ).to.equal( 2 );
+
+					expect( stubs.logger.error.callCount ).to.equal( 1 );
+					expect( stubs.logger.error.firstCall.args[ 0 ] ).to.equal( rejectError );
+
+					expect( stubs.logger.info.callCount ).to.equal( 10 );
+					expect( stubs.logger.info.getCall( 0 ).args[ 0 ] )
+						.to.equal( 'Updating dependencies for "@ckeditor/alpha"...' );
+					expect( stubs.logger.info.getCall( 1 ).args[ 0 ] )
+						.to.equal( 'Updating dependencies for "@ckeditor/beta"...' );
+					expect( stubs.logger.info.getCall( 2 ).args[ 0 ] )
+						.to.equal( 'Bumping version for "@ckeditor/alpha"...' );
+					expect( stubs.logger.info.getCall( 3 ).args[ 0 ] )
+						.to.equal( 'Bumping version for "@ckeditor/beta"...' );
+					expect( stubs.logger.info.getCall( 4 ).args[ 0 ] )
+						.to.equal( 'Pushing a local repository into the remote for "@ckeditor/alpha"...' );
+					expect( stubs.logger.info.getCall( 5 ).args[ 0 ] )
+						.to.equal( 'Pushing a local repository into the remote for "@ckeditor/beta"...' );
+					expect( stubs.logger.info.getCall( 6 ).args[ 0 ] )
+						.to.equal( 'Creating a GitHub release for "@ckeditor/alpha"...' );
+					expect( stubs.logger.info.getCall( 7 ).args[ 0 ] )
+						.to.equal( 'Cannot create a release on GitHub. Skipping that package.' );
+					expect( stubs.logger.info.getCall( 8 ).args[ 0 ] )
+						.to.equal( 'Creating a GitHub release for "@ckeditor/beta"...' );
+					expect( stubs.logger.info.getCall( 9 ).args[ 0 ] )
+						.to.equal( 'Created the release: https://github.com/ckeditor/ckeditor5-test-beta/releases/tag/v0.2.1' );
+				} );
+		} );
 	} );
 } );
 

